internal/auth0/resourceserver/resource_scope.go

```diff
@@ -0,0 +1,219 @@
+	api := meta.(*config.Config).GetAPI()
+
+	resourceServerID := data.Get("resource_server_identifier").(string)
+	scope := data.Get("scope").(string)
+	newDescription := data.Get("description").(string)
+
+	existingScopes, err := api.ResourceServer.Read(resourceServerID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	updatedScopes := []management.ResourceServerScope{}
+
+	found := false
+	for _, p := range existingScopes.GetScopes() {
+		updated := p
+		if p.GetValue() == scope {
+			found = true
+			updated.Description = &newDescription
+		}
+		updatedScopes = append(updatedScopes, updated)
+	}
+
+	if !found {
+		data.SetId("")
+		return nil
+	}
+
+	if err := api.ResourceServer.Update(resourceServerID, &management.ResourceServer{
+		Scopes: &updatedScopes,
+	}); err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(fmt.Sprintf(`%s::%s`, resourceServerID, scope))
+
+	return readResourceServerScope(ctx, data, meta)
+}
+
+func readResourceServerScope(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	resourceServerID := data.Get("resource_server_identifier").(string)
+	scope := data.Get("scope").(string)
+
+	existingScopes, err := api.ResourceServer.Read(resourceServerID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	for _, p := range existingScopes.GetScopes() {
+		if p.GetValue() == scope {
+			result := multierror.Append(
+				data.Set("description", p.GetDescription()),
+			)
+			return diag.FromErr(result.ErrorOrNil())
+		}
+	}
+
+	data.SetId("")
+	return nil
+}
+
+func deleteResourceServerScope(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	resourceServerID := data.Get("resource_server_identifier").(string)
+	scope := data.Get("scope").(string)
+
+	existingScopes, err := api.ResourceServer.Read(resourceServerID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	updateScopes := []management.ResourceServerScope{}
+	for _, p := range existingScopes.GetScopes() {
+		if p.GetValue() != scope {
+			updateScopes = append(updateScopes, p)
+		}
+	}
+
+	if err := api.ResourceServer.Update(
+		resourceServerID,
+		&management.ResourceServer{
+			Scopes: &updateScopes,
+		},
+	); err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	data.SetId("")
+	return nil
+}
+
+func importResourceServerScope(
```

Comment:
This seems to get repeated a lot, perhaps it's time to refactor the import logic? Or shall we defer until we consolidate the separators?